index.bs

```diff
@@ -164,16 +167,50 @@ In order to <dfn>create new anchor object</dfn> from |native origin| and |sessio
     1. Return |anchor|.
 </div>
 
+<div class="algorithm" data-algorithm="restore-anchor-from-frame">
+The {{XRAnchor/requestPersistentHandle()}} method, when invoked on an {{XRAnchor}} |anchor|, MUST run the following steps:
+    1. Let |promise| be [=a new Promise=] in the [=relevant realm=] of this {{XRSystem}}.
+    1. If |anchor|'s [=XRAnchor/deleted=] is <code>true</code>, [=/reject=] |promise| with {{InvalidStateError}}, return |promise|, and abort these steps.
+    1. Let |session| be |anchor|'s [=XRAnchor/session=].
+    1. If |session|'s [=XRSession/map of persistent anchors=] contains the value |anchor|, run the following steps:
+        1. Let |uuid| be the key of the mapping of the |anchor| value on |session|'s [=XRSession/map of persistent anchors=].
+        1. [=/Resolve=] |promise| with |uuid|.
+        1. Return |promise|.
+        1. Abort these steps.
+    1. Let |uuid| be the empty string.
+    1. Generate a UUID [[!RFC4122]] as a string and append it to |uuid|.
+    1. Add |uuid| and |anchor| to |session|'s [=XRSession/map of persistent anchors=].
+    1. [=/Resolve=] |promise| with |uuid|.
+    1. Return |promise|.
+</div>
+
+Note: when the {{XRSystem}} creates the unique ID, it MUST be unique and only known to the current origin.
+
+<div class="algorithm" data-algorithm="delete-peristence-from-anchor">
+The {{XRAnchor/deletePersistence()}} method, when invoked on an {{XRAnchor}} |anchor|, MUST run the following steps:
+    1. Let |promise| be [=a new Promise=] in the [=relevant realm=] of this {{XRSystem}}.
+    1. If |anchor|'s [=XRAnchor/deleted=] is <code>true</code>, [=/reject=] |promise| with {{InvalidStateError}}, return |promise|, and abort these steps.
+    1. Let |session| be |anchor|'s [=XRAnchor/session=].
+    1. If |session|'s [=XRSession/map of persistent anchors=] does not contain a mapping to |anchor|, [=/reject=] |promise| with {{InvalidStateError}}, return |promise|, and abort these steps.
+    1. Remove the mapping from |session|'s [=XRSession/map of persistent anchors=].
+    1. Return |promise|.
+</div>
+
+TODO: clarify this.
+
 Anchor creation {#anchor-creation}
 ===============
 
 The {{XRSession}} is extended to contain an associated <dfn for=XRSession>set of tracked anchors</dfn>.
 
+The {{XRSession}} is extended to contain an associated <dfn for=XRSession>map of persistent anchors</dfn> that is keyed with a UUID string and maps to an {{XRAnchor}}.
+
 The {{XRSession}} is extended to contain an associated <dfn for=XRSession>map of new anchors</dfn> that is keyed with {{XRAnchor}} object and maps to <code>{{Promise}}&lt;{{XRAnchor}}&gt;</code> object.
 
 <script type="idl">
 partial interface XRFrame {
   Promise<XRAnchor> createAnchor(XRRigidTransform pose, XRSpace space);
+  Promise<XRAnchor> restorePersistentAnchor(DOMString uuid);
```

Comment:
I _think_ it may be better suited to expose this on an XRSession - is there anything in the current XRFrame that we need in order to resolve a persisted anchor?

Reply:
I agree. I will make that change.